src/components/run-button/run-button-hooks.js

```diff
@@ -0,0 +1,175 @@
+    fetcher,
+    invalidations,
+    resultConversion,
+    runButtonType
+) {
+    const nodeUuidRef = useRef();
+    const studyUpdatedForce = useSelector((state) => state.studyUpdated);
+    const lastUpdateRef = useRef();
+    const dispatch = useDispatch();
+
+    const update = useCallback(() => {
+        nodeUuidRef.current = nodeUuid;
+        fetcher(studyUuid, nodeUuid)
+            .then((res) => {
+                if (nodeUuidRef.current === nodeUuid) {
+                    dispatch(
+                        setRunButtonStatus(
+                            runButtonType,
+                            resultConversion ? resultConversion(res) : res
+                        )
+                    );
+                }
+            })
+            .catch(() => {
+                dispatch(
+                    setRunButtonStatus(runButtonType, RunningStatus.FAILED)
+                );
+            });
+    }, [
+        runButtonType,
+        nodeUuid,
+        fetcher,
+        studyUuid,
+        resultConversion,
+        dispatch,
+    ]);
+
+    /* initial fetch and update */
+    useEffect(() => {
+        if (!studyUuid || !nodeUuid) {
+            return;
+        }
+
+        const isUpdateForUs = isWorthUpdate(
+            studyUpdatedForce,
+            fetcher,
+            lastUpdateRef,
+            nodeUuidRef,
+            nodeUuid,
+            invalidations
+        );
+        lastUpdateRef.current = { studyUpdatedForce, fetcher };
+        if (nodeUuidRef.current !== nodeUuid || isUpdateForUs) {
+            update();
+        }
+    }, [
+        update,
+        fetcher,
+        nodeUuid,
+        invalidations,
+        studyUpdatedForce,
+        studyUuid,
+    ]);
+}
+
+export function useNodeData(
+    studyUuid,
+    nodeUuid,
+    fetcher,
+    invalidations,
+    resultConversion
+) {
+    const [result, setResult] = useState();
+    const [isPending, setIsPending] = useState(false);
+    const [errorMessage, setErrorMessage] = useState(undefined);
+    const nodeUuidRef = useRef();
+    const studyUpdatedForce = useSelector((state) => state.studyUpdated);
+    const lastUpdateRef = useRef();
+
+    const update = useCallback(() => {
+        nodeUuidRef.current = nodeUuid;
+        setIsPending(true);
+        setErrorMessage(undefined);
+        fetcher(studyUuid, nodeUuid)
+            .then((res) => {
+                if (nodeUuidRef.current === nodeUuid) {
+                    setResult(resultConversion ? resultConversion(res) : res);
+                }
+            })
+            .catch((err) => {
+                setErrorMessage(err.message);
+                setResult(RunningStatus.FAILED);
+            })
+            .finally(() => setIsPending(false));
+    }, [nodeUuid, fetcher, studyUuid, resultConversion]);
+
+    /* initial fetch and update */
+    useEffect(() => {
+        if (!studyUuid || !nodeUuid) {
+            return;
+        }
+
+        const isUpdateForUs = isWorthUpdate(
```

Comment:
Do you understand why it was called `isUpdateForUs` ? I'm not sure and I don't find it very clear

Reply:
this hook will receive all the notifications targetting gridstudy
this is filtering whether this hook should update or not (same node ? same analysis type ? ...)